evennia/contrib/grid/ingame_map_display/ingame_map_display.py

```diff
@@ -0,0 +1,322 @@
+            room (Room) - The room to draw out
+            max_distance (int) - How extensive the map is
+        """
+        self.draw(room)
+        self.draw_exits(room)
+
+        if max_distance == 0:
+            return
+
+        # Check if the caller has access to the room in question. If not, don't draw it.
+        # Additionally, if the name of the exit is not ordinal but an alias of it is, use that.
+        for ex in [x for x in room.exits if x.access(self.caller, "traverse")]:
+            ex_name = self.exit_name_as_ordinal(ex)
+            if not ex_name or ex_name in ['up', 'down']:
+                continue
+            if self.has_drawn(ex.destination):
+                continue
+
+            self.update_pos(room, ex_name.lower())
+            self.draw_room_on_map(ex.destination, max_distance - 1)
+
+    def draw_exits(self, room):
+        """
+        Draw a given room's exit paths
+
+        Args:
+            room (Room) - The room to draw exits of
+        """
+        x, y = self.curX, self.curY
+        for ex in room.exits:
+            ex_name = self.exit_name_as_ordinal(ex)
+            if not ex_name:
+                continue
+
+            ex_character = _COMPASS_DIRECTIONS[ex_name][2]
+            delta_x = int(_COMPASS_DIRECTIONS[ex_name][1]/3)
+            delta_y = int(_COMPASS_DIRECTIONS[ex_name][0]/3)
+
+            # Make modifications if the exit has BOTH up and down exits
+            if ex_name == 'up':
+                if 'v' in self.grid[x][y]:
+                    self.render_room(room, x, y, p1='^', p2='v')
+                else:
+                    self.render_room(room, x, y, here='^')
+            elif ex_name == 'down':
+                if '^' in self.grid[x][y]:
+                    self.render_room(room, x, y, p1='^', p2='v')
+                else:
+                    self.render_room(room, x, y, here='v')
+            else:
+                self.grid[x + delta_x][y + delta_y] = ex_character
+
+    def draw(self, room):
+        """
+        Draw the map starting from a given room and add it to the cache of mapped rooms
+
+        Args:
+            room (Room) - The room to render
+        """
+        # draw initial caller location on map first!
+        if room == self.location:
+            self.start_loc_on_grid(room)
+            self.has_mapped[room] = [self.curX, self.curY]
+        else:
+            # map all other rooms
+            self.has_mapped[room] = [self.curX, self.curY]
+            self.render_room(room, self.curX, self.curY)
+
+    def render_room(self, room, x, y, p1='[', p2=']', here=None):
+        """
+        Draw a given room with ascii characters
+
+        Args:
+            room (Room) - The room to render
+            x (int) - The x-value of the room on the grid (horizontally, east/west)
+            y (int) - The y-value of the room on the grid (vertically, north/south)
+            p1 (char) - The first character of the 3-character room depiction
```

Comment:
Ther is no such thing as a `char` type, the `chr` type is something else. These are `str`.